src/mappings/core.ts

```diff
@@ -604,25 +604,38 @@ export function handlePairTokenPrice(block: ethereum.Block): void {
     return
   }
 
-  const fiveMinutes = BigInt.fromI32(5 * 60)
-  const fifteenMinutes = BigInt.fromI32(15 * 60)
-  const oneHour = BigInt.fromI32(60 * 60)
-  const twelveHours = BigInt.fromI32(12 * 60 * 60)
-
-  if (block.timestamp.mod(fiveMinutes).isZero()) {
-    createPairTokenPrice(block, pair, PairTokenPriceTimeframe.FIVE_MINUTES)
+  const blockTimestampDate = new Date(block.timestamp.toI64() * 1000)
+  const blockTimestampMinutes = blockTimestampDate.getUTCMinutes()
+  const blockTimestampHours = blockTimestampDate.getUTCHours()
+
+  const id = [
+    pair.id,
+    blockTimestampDate.getUTCFullYear().toString(),
+    blockTimestampDate.getUTCMonth().toString(),
+    blockTimestampHours.toString(),
+    blockTimestampMinutes.toString()
+  ].join('-')
+
+  const fiveMinutes = 5
+  const fifteenMinutes = 15
+  const twelveHours = 12
+
+  createPairTokenPrice(id, block, pair, PairTokenPriceTimeframe.ONE_MINUTE)
```

Comment:
I think this function is called on every block — every ~12 seconds after the merge. So, we overwrite data. If it's not a big issue for you, then merge it. The rest of code looks great.

Reply:
It's not a big issue I think. This will write the last block from that particular minute. This can be useful in the future.